src/main/java/edu/hm/hafner/analysis/parser/ClangAnalyzerPlistParser.java

```diff
@@ -0,0 +1,122 @@
+
+            XPathExpression diagDescriptionPath = compileDiagStrPath(xPath, "description");
+            XPathExpression diagCategoryPath = compileDiagStrPath(xPath, "category");
+            XPathExpression diagTypePath = compileDiagStrPath(xPath, "type");
+            XPathExpression diagLocationLinePath = compileDiagLocationPath(xPath, "line");
+            XPathExpression diagLocationColPath = compileDiagLocationPath(xPath, "col");
+            XPathExpression diagLocationFilePath = compileDiagLocationPath(xPath, "file");
+
+
+            IssueBuilder issueBuilder = new IssueBuilder();
+            Report report = new Report();
+
+            List<String> filesList = new ArrayList<>();
+
+            // Fill files list
+
+            NodeList files = (NodeList)filesPath.evaluate(doc, XPathConstants.NODESET);
+            for (Element filePathStr : XmlElementUtil.nodeListToList(files)) {
+                filesList.add(filePathStr.getTextContent());
+            }
+
+            // Parse errors
+
+            NodeList diagnostics = (NodeList)diagnosticsPath.evaluate(doc, XPathConstants.NODESET);
+            for (Element diag : XmlElementUtil.nodeListToList(diagnostics)) {
+                issueBuilder
+                    .setFileName(filesList.get(extractIntField(diag, diagLocationFilePath)))
+                    .guessSeverity("Warning") //.guessSeverity(extractSeverity(problem))
+                    .setMessage(extractField(diag, diagDescriptionPath))
+                    .setLineStart(extractIntField(diag, diagLocationLinePath))
+					.setColumnStart(extractIntField(diag, diagLocationColPath))
+                    .setCategory(extractField(diag, diagCategoryPath))
+                    .setType(extractField(diag, diagTypePath))
+                    //.setAdditionalProperties()
+                    ;
+
+                report.add(issueBuilder.build());
+            }
+
+            return report;
+        }
+        catch (XPathExpressionException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    private static XPathExpression compileDiagStrPath(final XPath xPath, final String field) throws XPathExpressionException
+    {
+        String search = "./key[text()='" + field + "']/following-sibling::string";
+        return xPath.compile(search);
+    }
+
+
+    private static XPathExpression compileDiagLocationPath(final XPath xPath, final String field) throws XPathExpressionException
+    {
+        String search = "./key[text()='location']/following-sibling::dict/key[text()='" + field + "']/following-sibling::integer";
+        return xPath.compile(search);
+    }
+
+
+    private static String extractField(final Element diag, final XPathExpression expr) throws XPathExpressionException
+    {
+        NodeList keys = (NodeList)expr.evaluate(diag, XPathConstants.NODESET);
+		String res = "";
+
+        for (Element k : XmlElementUtil.nodeListToList(keys)) {
+            res = k.getTextContent();
+			break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using a branching statement as the last in a loop.](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4824671)